theseus/geometry/functional/so3.py

```diff
@@ -0,0 +1,171 @@
+    # Compute the approximations when theta ~ 0
+    near_zero = theta < constants._SO3_NEAR_ZERO_EPS[tangent_vector.dtype]
+    non_zero = torch.ones(1, dtype=tangent_vector.dtype, device=tangent_vector.device)
+    theta_nz = torch.where(near_zero, non_zero, theta)
+    theta2_nz = torch.where(near_zero, non_zero, theta2)
+    sine = theta.sin()
+    cosine = torch.where(near_zero, 8 / (4 + theta2) - 1, theta.cos())
+    sine = theta.sin()
+    sine_by_theta = torch.where(near_zero, 0.5 * cosine + 0.5, sine / theta_nz)
+    one_minus_cosine_by_theta2 = torch.where(
+        near_zero, 0.5 * sine_by_theta, (1 - cosine) / theta2_nz
+    )
+    theta3_nz = theta_nz * theta2_nz
+    theta_minus_sine_by_theta3 = torch.where(
+        near_zero, torch.zeros_like(theta), (theta - sine) / theta3_nz
+    )
+
+    ret = (
+        one_minus_cosine_by_theta2
+        * tangent_vector.view(-1, 3, 1)
+        @ tangent_vector.view(-1, 1, 3)
+    )
+    ret[:, 0, 0] += cosine.view(-1)
+    ret[:, 1, 1] += cosine.view(-1)
+    ret[:, 2, 2] += cosine.view(-1)
+    sine_axis = sine_by_theta.view(-1, 1) * tangent_vector
+    ret[:, 0, 1] -= sine_axis[:, 2]
+    ret[:, 1, 0] += sine_axis[:, 2]
+    ret[:, 0, 2] += sine_axis[:, 1]
+    ret[:, 2, 0] -= sine_axis[:, 1]
+    ret[:, 1, 2] -= sine_axis[:, 0]
+    ret[:, 2, 1] += sine_axis[:, 0]
+
+    jac = (
+        theta_minus_sine_by_theta3
+        * tangent_vector.view(-1, 3, 1)
+        @ tangent_vector.view(-1, 1, 3)
+    )
+    diag_jac = jac.diagonal(dim1=1, dim2=2)
+    diag_jac += sine_by_theta.view(-1, 1)
+    jac_temp = one_minus_cosine_by_theta2.view(-1, 1) * tangent_vector
+    jac[:, 0, 1] += jac_temp[:, 2]
+    jac[:, 1, 0] -= jac_temp[:, 2]
+    jac[:, 0, 2] -= jac_temp[:, 1]
+    jac[:, 2, 0] += jac_temp[:, 1]
+    jac[:, 1, 2] += jac_temp[:, 0]
+    jac[:, 2, 1] -= jac_temp[:, 0]
+
+    return [jac], ret
+
+
+class ExpMap(LieGroup.UnaryOperator):
+    @classmethod
+    def forward(cls, ctx, tangent_vector):
+        tangent_vector: torch.Tensor = cast(torch.Tensor, tangent_vector)
+        ret = _exp_map_impl(tangent_vector)
+        ctx.save_for_backward(tangent_vector, ret)
+        return ret
+
+    @classmethod
+    def backward(cls, ctx, grad_output):
+        tangent_vector: torch.Tensor = ctx.saved_tensors[0]
+        group: torch.Tensor = ctx.saved_tensors[1]
+        if not hasattr(ctx, "jacobians"):
+            ctx.jacobians: torch.Tensor = _jexp_map_impl(tangent_vector)[0][0]
+        jacs = ctx.jacobians
+        dR = group.transpose(1, 2) @ grad_output
+        grad_input = jacs.transpose(1, 2) @ torch.stack(
+            (
+                dR[:, 2, 1] - dR[:, 1, 2],
+                dR[:, 0, 2] - dR[:, 2, 0],
+                dR[:, 1, 0] - dR[:, 0, 1],
+            ),
+            dim=1,
+        ).view(-1, 3, 1)
+        return grad_input.view(-1, 3)
+
+
+_module = get_module(__name__)
+
+_exp_map_autograd_fn = ExpMap.apply
+_jexp_map_autograd_fn = _jexp_map_impl
```

Comment:
Add a comment here that says we are using this as a placeholder while we add an actual implementation for this.

Reply:
Fixed.